packages/devtools/lib/src/timeline/timeline_protocol.dart

```diff
@@ -326,6 +370,29 @@ class TimelineData {
       frame.addedToTimeline = true;
     }
   }
+
+  TimelineEventType _inferEventType(TraceEvent event) {
+    if (event.threadId == cpuThreadId) {
+      return TimelineEventType.cpu;
+    } else if (event.threadId == gpuThreadId) {
+      return TimelineEventType.gpu;
+    } else {
+      return TimelineEventType.unknown;
+    }
+  }
+
+  bool _shouldProcessTraceEvent(TraceEvent event) {
+    const List<String> phaseWhitelist = ['s', 'f', 'B', 'E', 'X'];
```

Comment:
this would be faster as a Set literal  \o/

Reply:
done.